lib/interceptor.js

```diff
@@ -196,7 +196,9 @@ Interceptor.prototype.reqheaderMatches = function reqheaderMatches(
   const reqHeader = this.reqheaders[key]
   let header = options.headers[key]
 
-  // headers on the request should only be strings or arrays
+  // options.headers are pulled straight from ClientRequest so a header field value should
+  // be a string or an array of strings eg. `set-cookie`, however, nothing enforces those types.
+  // For the ease of comparing, we cast all values to strings.
   if (header && typeof header !== 'string' && header.toString) {
     header = header.toString()
   }
```

Comment:
Can you think of a case where header

1. truthy
2. not a string
3. but does _not_ have a toString method?

I wonder if we could simplify the code to something like

```js
header = String(header || '')
```

or at least

```js
if (header) {
  header = String(header)
}
```

Reply:
I think both of those options would work. The check being done is also being done inside `matchStringOrRegexp`. It's just been duplicated here so any user provided cb gets the string version. 

Adding this comment in this PR may have been poorly thought out on my end as I'm planning on doing more header matching work in the near future. Especially once #1632 is merged in. 

I think I would feel better if I removed this comment from this PR and just included it in my other work, yet to come.

Reply:
Okay, sounds good to me